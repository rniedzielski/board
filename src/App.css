.App {
  text-align: center;
  width: 300px;
  margin: 10px auto;
  background: #eee;
  border: 1px solid #e0e0e0;
  padding: 10px;
}

.WhillBox {
  margin: 20px auto;
  display: flex;
  padding: 10px;
  border: 1px solid #e0e0e0;
  width: 250px;
  justify-content: space-between;
  background: #fff;
  box-shadow: inset 0 0 4px #000;
}

.Buttons {
  margin: 20px auto;
  display: flex;
  width: 250px;
  justify-content: space-around;
}

.Buttons button {
  background: #6daa1c;
  font-size: 16px;
  padding: 5px 25px;
}

.Buttons [disabled] {
  background: rgba(109, 170, 28, .4)
}
