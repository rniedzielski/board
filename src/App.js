import React, { Component } from 'react';
import './App.css';
import Buttons from './components/Buttons/Buttons';
import WhillBox from './components/WhillBox/WhillBox';

import strawberry from './assets/strawberry.png';
import banana from './assets/banana.png';
import orange from './assets/orange.png';
import monkey from './assets/monkey.png';

class App extends Component {
    state = {
        pictures: [strawberry, banana, orange, monkey],
        selected: [],
        interval: null,
        timeout: null,
        totalAmount: 0,
        stopAction: false
    }

    constructor(props) {
        super(props)
        this.generateRandomPhotos();
    }

    startDrawHandler = () => {
        let time = new Date();
        this.state.interval = setInterval(() => {
            let currentTime = new Date();

            this.generateRandomPhotos();

            this.setState({
                selected: this.state.selected, 
                interval: this.state.interval,
                stopAction: false
            });

            if (time.getTime() + 10000 < currentTime.getTime()) {
                clearInterval(this.state.interval);
                this.stopDrawHandler();
            }
        }, 50)
    }

    stopDrawHandler = () => {
        let selected = this.state.selected;
        if (selected[0] === selected[2] && selected[0] !== selected[1]) {
            this.state.totalAmount += 10;
        } else if ((selected[0] === selected[1] || selected[1] === selected[2]) && selected[0] !== selected[2]) {
            this.state.totalAmount += 20;
        } else if (selected[0] === selected[1] && selected[1] === selected[2]) {
            this.state.totalAmount += 100;
        } else {
            this.state.totalAmount = 0;
        }

        clearInterval(this.state.interval);
        clearTimeout(this.state.timeout);
        this.setState({selected: this.state.selected, totalAmount: this.state.totalAmount, stopAction: true})
        
    }

    generateRandomPhotos () {
        for (let i = 0; i < 3; i++) {
            this.state.selected[i] = Math.floor(Math.random() * this.state.pictures.length);
        }
    }

    render() {
        this.state.timeout = setTimeout(() => {
            if (this.state.interval === null) {
                this.startDrawHandler();
            }
        }, 5000)

        return (
            <div className="App">
                <WhillBox pictures={this.state.pictures} selected={this.state.selected} />
                <Buttons start={this.startDrawHandler} stop={this.stopDrawHandler} stopAction={this.state.stopAction} />
                <p>You win: <strong>{this.state.totalAmount} $</strong></p>
            </div>
        );
    }
}

export default App;
