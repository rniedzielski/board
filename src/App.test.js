import React from 'react';
import ReactDOM from 'react-dom';

import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'

configure({adapter: new Adapter()});

import App from './App';

describe('WhillBox', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('starts with a win text "You win: 0 $"', () => {
        const container = shallow(<App />);
        const text = container.find('p').text();

        expect(text).toEqual('You win: 0 $');
    });

    it('should Buttons elment be defined', () => {
        const container = shallow(<App />);

        expect(container.find('Buttons')).toBeDefined();
    });
});
