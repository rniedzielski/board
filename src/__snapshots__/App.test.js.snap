// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhillBox should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <whillBox
          pictures={
            Array [
              "strawberry.png",
              "banana.png",
              "orange.png",
              "monkey.png",
            ]
          }
          selected={
            Array [
              1,
              3,
              3,
            ]
          }
        />,
        <buttons
          start={[Function]}
          stop={[Function]}
          stopAction={false}
        />,
        <p>
          You win: 
          <strong>
            0
             $
          </strong>
        </p>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "pictures": Array [
            "strawberry.png",
            "banana.png",
            "orange.png",
            "monkey.png",
          ],
          "selected": Array [
            1,
            3,
            3,
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "start": [Function],
          "stop": [Function],
          "stopAction": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "You win: ",
            <strong>
              0
               $
            </strong>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "You win: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                0,
                " $",
              ],
            },
            "ref": null,
            "rendered": Array [
              0,
              " $",
            ],
            "type": "strong",
          },
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <whillBox
            pictures={
              Array [
                "strawberry.png",
                "banana.png",
                "orange.png",
                "monkey.png",
              ]
            }
            selected={
              Array [
                1,
                3,
                3,
              ]
            }
          />,
          <buttons
            start={[Function]}
            stop={[Function]}
            stopAction={false}
          />,
          <p>
            You win: 
            <strong>
              0
               $
            </strong>
          </p>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "pictures": Array [
              "strawberry.png",
              "banana.png",
              "orange.png",
              "monkey.png",
            ],
            "selected": Array [
              1,
              3,
              3,
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "start": [Function],
            "stop": [Function],
            "stopAction": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "You win: ",
              <strong>
                0
                 $
              </strong>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "You win: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  0,
                  " $",
                ],
              },
              "ref": null,
              "rendered": Array [
                0,
                " $",
              ],
              "type": "strong",
            },
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
